### PR TITLE
Updates Launchpad search UX to avoid always searching with the API

### LINKED ISSUE #3855: Launchpad PR search refinement
-Show an option to search for pull requests
-Will switch you to search mode and enable the search
-Do not use search api if not in that mode
-Pasting a PR url when not in search mode will put you in search mode
-Clearing your search when in search mode will restore your original items

### PR DESCRIPTION
**Pull request by axosoft-ramint**
Closes #3855 

Current approach has some hacky workarounds to accommodate:

- Directly moving into search mode when a PR url is pasted in
- Not losing the content of the search box when toggling search mode on
- Showing the option to toggle search mode on if no results are found when searching with it off

Suggestions welcome

### CODE REVIEW

File: src/plus/launchpad/launchpad.ts [hunk 1]
```diff
@@ -1510,3 +1591,15 @@
 function isLaunchpadTargetActionQuickPickItem(item: any): item is QuickPickItemOfT<LaunchpadTargetAction> {
 	return item?.item?.action != null && item?.item?.target != null;
 }
+
+function isGitHubPullRequestUrl(search: string) {
+	return search.includes('github.com') && search.includes('/pull/');
```
[the remark this answers is not in the document]
Comment (axosoft-ramint) [reply src/plus/launchpad/launchpad.ts]: Fixed in commit: [Improves PR url checks](https://github.com/gitkraken/vscode-gitlens/pull/3856/commits/d6e3a87b7dbbbb0ed474954d4cd46804187a0763)

File: src/plus/launchpad/launchpad.ts [hunk 2]
```diff
@@ -1510,3 +1591,15 @@
...
+}
+
+function isGitLabPullRequestUrl(search: string) {
+	return search.includes('gitlab.com') && search.includes('/merge_requests/');
```
[the remark this answers is not in the document]
Comment (axosoft-ramint) [reply src/plus/launchpad/launchpad.ts]: Fixed in commit: [Improves PR url checks](https://github.com/gitkraken/vscode-gitlens/pull/3856/commits/d6e3a87b7dbbbb0ed474954d4cd46804187a0763)

File: src/plus/launchpad/launchpad.ts [hunk 3]
```diff
@@ -568,86 +704,93 @@ export class LaunchpadCommand extends QuickCommand<State> {
...
+			// TODO@axosoft-ramint why is this needed?
+			onDidActivate: quickpick => {
+				if (!this.savedSearch?.length) return;
+
+				if (context.inSearch === 'mode') {
+					quickpick.value = this.savedSearch;
+				}
+
+				this.savedSearch = undefined;
+			},
```
Comment (eamodio) [inline src/plus/launchpad/launchpad.ts:746]: @axosoft-ramint Why is this needed?

**Review by eamodio — APPROVED** [review]
Pushed more experience changes
